code/__HELPERS/icon_smoothing.dm

```diff
@@ -41,18 +41,61 @@
 		if(!AM.anchored)
 			return 0
 
-		for(var/direction in alldirs)
+		for(var/direction in cardinal)
 			AM = find_type_in_direction(A, direction)
-			if(istype(AM))
-				if(AM.anchored)
-					adjacencies |= 1 << direction
-			else
-				if(AM)
-					adjacencies |= 1 << direction
+			if(istype(AM) && AM.anchored)
+				adjacencies |= 1 << direction
+			else if(AM)
+				adjacencies |= 1 << direction
```

Comment:
So we're counting on the fact that future coders will never give `can_be_unanchored = 1` to a non movable atom  ? Why is can_be_unanchored even defined at atom level to begin with? Move that var to /atom/movable and just put an istype(A,/atom/movable)) before checking can_be_unanchored. This way you can remove the `if(istype(AM))` below.